### PR TITLE
seccomp: filter openat2() entirely in seccomp_restrict_sxid()

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/shared/seccomp-util.c [hunk 1]
```diff
@@ -2002,6 +2002,22 @@ static int seccomp_restrict_sxid(scmp_filter_ctx seccomp, mode_t m) {
         else
                 any = true;
 
+#if SCMP_SYS(openat2) > 0
```
Comment (mrc0mmand) [inline src/shared/seccomp-util.c]:
```
    archlinux_systemd_ci: ../build/src/shared/seccomp-util.c:2005:5: error: '__SNR_openat2' is not defined, evaluates to 0 [-Werror,-Wundef]
    archlinux_systemd_ci: #if SCMP_SYS(openat2) > 0
    archlinux_systemd_ci:     ^
    archlinux_systemd_ci: /usr/include/seccomp.h:200:23: note: expanded from macro 'SCMP_SYS'
    archlinux_systemd_ci: #define SCMP_SYS(x)             (__SNR_##x)
    archlinux_systemd_ci:                                  ^
    archlinux_systemd_ci: <scratch space>:75:1: note: expanded from here
    archlinux_systemd_ci: __SNR_openat2
    archlinux_systemd_ci: ^
    archlinux_systemd_ci: 1 error generated.
```

**Comment by poettering** [comment]
Force pushed new version with issue @mrc0mmand pointed out fixed

File: src/shared/seccomp-util.c [hunk 2]
```diff
@@ -2002,6 +2002,22 @@ static int seccomp_restrict_sxid(scmp_filter_ctx seccomp, mode_t m) {
         else
                 any = true;
 
+#if defined(__SNR_openat2)
+        /* The new openat2() system call can't be filtered sensibly, since it moves the flags parameter into
+         * an indirect structure. Let's bock it entirely for now. That should be a reasonably OK thing to do
```
Comment (keszybz) [inline src/shared/seccomp-util.c]: bock → block

**Comment by poettering** [comment]
fixed the bock/block thing now, no other changes. upgrading green label

**Pull request merged**